honeycomb/honeycomb.go

```diff
@@ -334,19 +339,36 @@ type span struct {
 	HasRemoteParent bool    `json:"has_remote_parent"`
 }
 
-func getHoneycombTraceID(traceID string) string {
-	hcTraceUUID, err := uuid.Parse(traceID)
-	if err != nil {
-		return ""
+// getHoneycombTraceID returns a trace_id suitable for use in honeycomb. Before
+// encoding the bytes as a hex string, we want to handle cases where we are
+// given 128 bit ids with zero padding, e.g. 0000000000000000f798a1e7f33c8af6.
+// To do this, we borrow a strategy from Jaeger [1] wherein we split the byte
+// sequence into two parts. The leftmost part could contain all zeros. We use
+// that to determine whether to return a 64 bit hex encoded string or a 128 bit
+// one.
+//
+// [1]: https://github.com/jaegertracing/jaeger/blob/cd19b64413eca0f06b61d92fe29bebce1321d0b0/model/ids.go#L81
+func getHoneycombTraceID(traceID []byte) string {
+	var high, low uint64
+	if len(traceID) == traceIdLongLength {
+		high = binary.BigEndian.Uint64(traceID[:traceIdShortLength])
+		low = binary.BigEndian.Uint64(traceID[traceIdShortLength:])
+	} else {
+		low = binary.BigEndian.Uint64(traceID)
 	}
-	return hcTraceUUID.String()
+
+	if high == 0 {
```

Comment:
You can invert this condition and move it up to line 356, just after assigning `low`. For the alternate case at line 357, you know that `high` will be zero.

In other words:
```go
var low uint64
if len(traceID) == traceIdLongLength {
	low = binary.BigEndian.Uint64(traceID[traceIdShortLength:])
	if high := binary.BigEndian.Uint64(traceID[:traceIdShortLength]); high != 0 {
		return fmt.Sprintf("%016x%016x", high, low)
	}
} else {
	// ...
}
return fmt.Sprintf("%016x", low)
```